scripts/check_repo_hygiene.py

```
@@ -94,10 +94,90 @@ def check_hardcoded_paths(tracked: list[str]) -> list[str]:
+        line = line.split("#", 1)[0].strip()
+        if line:
+            globs.append(line)
+    return globs
+
+
+def _changed_files(base: str = "origin/main") -> tuple[list[str], str | None]:
+    """Files changed vs the merge base. Returns (files, skip_reason)."""
+    import subprocess
+
+    try:
+        mb = subprocess.run(["git", "merge-base", base, "HEAD"],
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)